src/reducers/profile-view.js

```diff
@@ -499,6 +447,96 @@ const rightClickedTrack: Reducer<TrackReference | null> = (
   }
 };
 
+const rightClickedCallNodePath: Reducer<RightClickedCallNodePath | null> = (
+  state = null,
+  action
+) => {
+  switch (action.type) {
+    case 'COALESCED_FUNCTIONS_UPDATE': {
+      const { functionsUpdatePerThread } = action;
+
+      if (state === null) {
+        return null;
+      }
+
+      const functionUpdate = functionsUpdatePerThread[state.threadIndex];
+
+      if (!functionUpdate) {
+        console.error(
```

Comment:
As mentioned in https://github.com/firefox-devtools/profiler/pull/2269#discussion_r344297102 I added a warning message and a state reset for this reducer when there is no `functionUpdate` for the thread found, but now I'm wondering should something like this also be applied to [other](https://github.com/firefox-devtools/profiler/blob/c75517fad718995ff3c87a53291a2fa32cf33140/src/reducers/profile-view.js#L45-L47) [places](https://github.com/firefox-devtools/profiler/blob/c75517fad718995ff3c87a53291a2fa32cf33140/src/reducers/profile-view.js#L143-L145) where reducer doesn't have a `functionUpdate` for the thread? Currently those uther reducers just keep the state intact if that happens

Reply:
I think I was wrong before: it's totally normal that we can't find a thread index.
You can look at the class `ColascedFunctionsUpdateDispatcher` in `src/actions/receive-profile`: this happens during load time, when we symbolicate the profile, and we find that some functions need to be merged (when 2 symbols A and symbols B resolve to the same function). This doesn't necessarily happen in all threads. (I don't know the exact situations when this happens or doesn't happen).

Reply:
You can also look at ` finishSymbolicationForLib` in `src/profile-logic/symbolication`.

Reply:
Ahh, yes, makes sense! I'll change it then to just return the current state if there is no `functionUpdate`